src/components/reviewMatches/DataGrid.tsx

```diff
@@ -0,0 +1,281 @@
+const getCellClassName = (
+  params: GridCellParams<string>,
+  field: DisplayField,
+  data: AnyRecord
+) => {
+  if (field.groups.includes('demographics')) {
+    return params.value === data[params.field] ? 'matching-cell' : ''
+  } else return ''
+}
+
+const DataGrid: React.FC<DataGridProps> = ({
+  data,
+  handleOpenLinkedRecordDialog,
+  handleOpenCreateNewGRDialog,
+  hideAction = false,
+  sx
+}) => {
+  const { availableFields } = useAppConfig()
+  const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null)
+
+  const isOpen = Boolean(anchorEl)
+  const open = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget)
+  }
+  const close = () => {
+    setAnchorEl(null)
+  }
+
+  const columns: GridColumns = [
+    ...availableFields.map(field => {
+      const { fieldName, fieldLabel, formatValue } = field
+      if (fieldName === 'recordType') {
+        return {
+          field: fieldName,
+          headerName: fieldLabel,
+          flex: 1,
+          valueFormatter: (
+            params: GridValueFormatterParams<number | string | Date>
+          ) => formatValue(params.value),
+          cellClassName: (params: GridCellParams<string>) =>
+            getRecordTypeClassName(params),
+          renderCell: (params: GridRenderCellParams) => {
+            switch (params.row.type) {
+              case 'Current':
+                return <Typography>Patient</Typography>
+              case 'Golden':
+                return (
+                  <Typography color="#D79B01" fontWeight={700}>
+                    Golden
+                  </Typography>
+                )
+              case 'Candidate':
+                if (params.row.searched) {
+                  return <Typography>Searched</Typography>
+                } else {
+                  return <Typography>Blocked</Typography>
+                }
+              default:
+                return <></>
+            }
+          }
+        }
+      }
+      if (fieldName === 'uid') {
+        return {
+          field: fieldName,
+          headerName: fieldLabel,
+          flex: 1,
+          renderCell: (params: GridRenderCellParams) => {
+            if (params.row.type === 'Current') {
+              return (
+                <Link href={`/patient-record/${params.row.uid}`}>
+                  {params.row.uid}
+                </Link>
+              )
+            }
+            return (
+              <Link href={`/golden-record/${params.row.uid}`}>
+                {params.row.uid}
+              </Link>
+            )
+          }
+        }
+      }
+      return {
+        field: fieldName,
+        headerName: fieldLabel,
+        flex: 1,
+        valueFormatter: (
+          params: GridValueFormatterParams<number | string | Date>
+        ) => formatValue(params.value),
+        cellClassName: (params: GridCellParams<string>) =>
+          getCellClassName(params, field, data[0])
+      }
+    }),
+    {
+      field: 'score',
```

Comment:
Score field could be dealt with using available fields. "score" field is part of the systemFields that gets returned by the API.